staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields_test.go

```diff
@@ -198,3 +200,113 @@ time: "2001-02-03T04:05:06Z"
 		})
 	}
 }
+
+func TestSortEncodedManagedFields(t *testing.T) {
+	tests := []struct {
+		name          string
+		managedFields []metav1.ManagedFieldsEntry
+		expected      []metav1.ManagedFieldsEntry
+	}{
+		{
+			name:          "empty",
+			managedFields: []metav1.ManagedFieldsEntry{},
+			expected:      []metav1.ManagedFieldsEntry{},
+		},
+		{
+			name:          "nil",
+			managedFields: nil,
+			expected:      nil,
+		},
+		{
+			name: "remains untouched",
+			managedFields: []metav1.ManagedFieldsEntry{
+				{Manager: "a", Operation: metav1.ManagedFieldsOperationApply, Time: nil},
+				{Manager: "a", Operation: metav1.ManagedFieldsOperationUpdate, Time: parseTimeOrPanic("2001-01-01T01:00:00Z")},
+			},
+			expected: []metav1.ManagedFieldsEntry{
+				{Manager: "a", Operation: metav1.ManagedFieldsOperationApply, Time: nil},
+				{Manager: "a", Operation: metav1.ManagedFieldsOperationUpdate, Time: parseTimeOrPanic("2001-01-01T01:00:00Z")},
+			},
+		},
+		{
+			name: "apply first",
+			managedFields: []metav1.ManagedFieldsEntry{
+				{Manager: "a", Operation: metav1.ManagedFieldsOperationUpdate, Time: parseTimeOrPanic("2001-01-01T01:00:00Z")},
+				{Manager: "a", Operation: metav1.ManagedFieldsOperationApply, Time: nil},
+			},
+			expected: []metav1.ManagedFieldsEntry{
+				{Manager: "a", Operation: metav1.ManagedFieldsOperationApply, Time: nil},
+				{Manager: "a", Operation: metav1.ManagedFieldsOperationUpdate, Time: parseTimeOrPanic("2001-01-01T01:00:00Z")},
+			},
+		},
+		{
+			name: "newest first",
+			managedFields: []metav1.ManagedFieldsEntry{
+				{Manager: "c", Operation: metav1.ManagedFieldsOperationUpdate, Time: parseTimeOrPanic("2001-01-01T01:00:00Z")},
+				{Manager: "a", Operation: metav1.ManagedFieldsOperationApply, Time: nil},
+				{Manager: "c", Operation: metav1.ManagedFieldsOperationUpdate, Time: parseTimeOrPanic("2002-01-01T01:00:00Z")},
+				{Manager: "b", Operation: metav1.ManagedFieldsOperationApply, Time: nil},
+			},
+			expected: []metav1.ManagedFieldsEntry{
+				{Manager: "a", Operation: metav1.ManagedFieldsOperationApply, Time: nil},
+				{Manager: "b", Operation: metav1.ManagedFieldsOperationApply, Time: nil},
+				{Manager: "c", Operation: metav1.ManagedFieldsOperationUpdate, Time: parseTimeOrPanic("2002-01-01T01:00:00Z")},
+				{Manager: "c", Operation: metav1.ManagedFieldsOperationUpdate, Time: parseTimeOrPanic("2001-01-01T01:00:00Z")},
+			},
+		},
+		{
+			name: "manager last",
```

Comment:
is this test really comparing the manager since they have a different time anyway?